code/modules/surgery/organs/liver.dm

```diff
@@ -118,3 +118,21 @@
 		COOLDOWN_START(src, severe_cooldown, 10 SECONDS)
 	if(prob(emp_vulnerability/severity)) //Chance of permanent effects
 		organ_flags |= ORGAN_SYNTHETIC_EMP //Starts organ faliure - gonna need replacing soon.
+
+/obj/item/organ/liver/cybernetic/upgraded/ipc
+	name = "substance processor"
+	icon_state = "substance_processor"
+	attack_verb = list("processed")
+	desc = "A machine component, installed in the chest. This grants the Machine the ability to process chemicals that enter its systems."
+	alcohol_tolerance = 0
+	toxTolerance = -1
+	toxLethality = 0
+	status = ORGAN_ROBOTIC
+
+/obj/item/organ/liver/cybernetic/upgraded/ipc/emp_act(severity)
```

Comment:
damn, sounds like it's in your best interesting to just ditch this thing near-roundstart as an IPC; you can't metabolize most chems anyway, and welders and cable coils are incredibly easy to come by.